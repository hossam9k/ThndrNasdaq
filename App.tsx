import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import Tickers from './src/screens/Tickers';
import colors from './src/styles/colors';
import Header from './src/components/Header';
import SplashScreen from './src/screens/SplashScreen';

function App(): React.JSX.Element {
  const [isSplashVisible, setSplashVisible] = React.useState(true);

  // Simulate a loading delay (e.g., 3 seconds)
  React.useEffect(() => {
    const timer = setTimeout(() => {
      setSplashVisible(false);
    }, 3000); // 3 seconds delay for the splash screen

    return () => clearTimeout(timer); // Clear timer if the component unmounts
  }, []);

  // Show splash screen or main app content
  return isSplashVisible ? (
    <SplashScreen />
  ) : (
    <View style={styles.outerContainer}>
      <View style={styles.headerSpacing}></View>
      <>
        <Header />
        <Tickers />
      </>
    </View>
  );
}

const styles = StyleSheet.create({
  outerContainer: {
    flex: 1,
    backgroundColor: colors.darkBlue,
  },
  headerSpacing: {
    paddingBottom: 40,
    backgroundColor: colors.darkBlue2,
  },
});

export default App;