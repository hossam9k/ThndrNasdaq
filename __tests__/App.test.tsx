import React from 'react';
import { render, screen } from '@testing-library/react-native';
import App from '../App'; // Adjust the import path if necessary

describe('App Component', () => {
  it('renders correctly', () => {
    render(<App />);
    // Add assertions to test your component
    expect(screen.toJSON()).toMatchSnapshot();  // For snapshot testing
  });
});