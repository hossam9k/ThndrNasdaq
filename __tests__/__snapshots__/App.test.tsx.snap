// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#212138",
      "flex": 1,
      "justifyContent": "space-evenly",
      "padding": 20,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "testUri": "../../../assets/file.png",
      }
    }
    style={
      {
        "height": 933.8,
        "width": 525,
      }
    }
  />
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#ffffff",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Thundr
  </Text>
</View>
`;
